common/src/main/java/com/wynntils/screens/maps/IconFilterScreen.java

```diff
@@ -0,0 +1,285 @@
+
+        this.addRenderableWidget(excludeAllButton);
+        // endregion
+
+        // region done button
+        this.addRenderableWidget(
+                new Button.Builder(Component.translatable("screens.wynntils.iconFilter.done"), (button) -> McUtils.mc()
+                                .setScreen(previousScreen))
+                        .pos((int) (dividedWidth * 44), (int) (dividedHeight * 3))
+                        .size(filterButtonWidth, 20)
+                        .build());
+        // endregion
+
+        // The icons that will be displayed
+        usedIcons = new ArrayList<>(icons.keySet());
+
+        populateIcons();
+    }
+
+    @Override
+    public void doRender(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        renderBackground(guiGraphics, mouseX, mouseY, partialTick);
+        renderScrollButton(guiGraphics.pose());
+        super.doRender(guiGraphics, mouseX, mouseY, partialTick);
+    }
+
+    @Override
+    public void renderBackground(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        super.renderBackground(guiGraphics, mouseX, mouseY, partialTick);
+
+        RenderUtils.drawScalingTexturedRect(
+                guiGraphics.pose(),
+                Texture.WAYPOINT_MANAGER_BACKGROUND.resource(),
+                backgroundX,
+                backgroundY,
+                0,
+                backgroundWidth,
+                backgroundHeight,
+                Texture.WAYPOINT_MANAGER_BACKGROUND.width(),
+                Texture.WAYPOINT_MANAGER_BACKGROUND.height());
+    }
+
+    public void toggleIcon(Texture icon) {
+        // Toggle the icon
+        icons.put(icon, !icons.get(icon));
+
+        previousScreen.setFilteredIcons(icons);
+
+        // Only have the buttons active if they will do anything.
+        // Eg. Include all will set all to true, so deactive it if no falses
+        includeAllButton.active = icons.containsValue(false);
+        excludeAllButton.active = icons.containsValue(true);
+
+        populateIcons();
+    }
+
+    @Override
+    public boolean doMouseClicked(double mouseX, double mouseY, int button) {
+        for (GuiEventListener child : children()) {
+            if (child.isMouseOver(mouseX, mouseY)) {
+                return child.mouseClicked(mouseX, mouseY, button);
+            }
+        }
+
+        if (!draggingScroll && (usedIcons.size() > MAX_ICONS_TO_DISPLAY)) {
+            float scrollButtonRenderX = (int) (dividedWidth * 52);
+
+            if (mouseX >= scrollButtonRenderX
+                    && mouseX <= scrollButtonRenderX + dividedWidth
+                    && mouseY >= scrollButtonRenderY
+                    && mouseY <= scrollButtonRenderY + scrollButtonHeight) {
+                draggingScroll = true;
+            }
+        }
+
+        // Returning super.doMouseClicked(mouseX, mouseY, button) causes ConcurrentModificationException
+        return true;
```

Comment:
Doesn't seem to happen to me, either way, I don't think this comment is needed.

Reply:
Maybe 1.20.2 fixed it somehow. Only left it in incase someone knew how to fix it

Reply:
Hm, okay. It does happen. It happens because when you click the include/exclude all buttons, you repopulate all widgets, while iterating over then. But I don't think you need to call this super, you do everything needed above.